### PR TITLE
feat: Update NumericalSolver::StateVector to Eigen VectorXd

### PR DESCRIPTION
**Pull request by vishwa2710**

### CODE REVIEW

File: src/OpenSpaceToolkit/Astrodynamics/Flight/System/Dynamics.cpp [hunk 1]
```diff
@@ -58,7 +58,7 @@ void Dynamics::DynamicalEquations(
     const Instant& anInstant
 )
 {
-    for (Index i = 0; i < dxdt.size(); ++i)
+    for (Index i = 0; i < (Size)dxdt.size(); ++i)
```
Comment (kyle-cochran) [inline src/OpenSpaceToolkit/Astrodynamics/Flight/System/Dynamics.cpp:61]:
```suggestion
    for (Size i = 0; i < (Size)dxdt.size(); ++i)
```

I know they're the same under the hood, but this looks a bit nicer IMO

**Review by kyle-cochran — APPROVED** [review]
One nit suggestion, otherwise lgtm

File: src/OpenSpaceToolkit/Astrodynamics/Flight/System/Dynamics.cpp (hunk 1 above)
Comment (Derollez) [reply src/OpenSpaceToolkit/Astrodynamics/Flight/System/Dynamics.cpp:61]: Agreed!

**Review by Derollez — APPROVED** [review]

File: src/OpenSpaceToolkit/Astrodynamics/Flight/System/Dynamics.cpp (hunk 1 above)
Comment (vishwa2710) [reply src/OpenSpaceToolkit/Astrodynamics/Flight/System/Dynamics.cpp:61]: Agreed as well, Thanks for thinking this over!

**Pull request merged**